sql/analyzer/validation_rules.go

```diff
@@ -320,16 +320,16 @@ func expressionReferencesOnlyGroupBys(groupBys []string, expr sql.Expression) bo
 		case nil, sql.Aggregation, *expression.Literal:
 			return false
 		case *expression.Alias, sql.FunctionExpression:
-			if stringContains(groupBys, expr.String()) {
+			if stringContains(groupBys, strings.ToLower(expr.String())) {
```

Comment:
might be safe to just always lower the target in `stringContains`, there are other uses of the function that might not be airtight